lib/linguist/heuristics.rb

```diff
@@ -15,12 +15,15 @@ class Heuristics
     # Returns an Array of languages, or empty if none matched or were inconclusive.
     def self.call(blob, languages)
       data = blob.data
+      language_names = languages.map(&:name)
+      result = nil
 
       @heuristics.each do |heuristic|
-        return Array(heuristic.call(data)) if heuristic.matches?(languages)
+        result = heuristic.call(data) if heuristic.matches?(language_names)
+        break if result
```

Comment:
These changes are subtle, but now it will keep trying heuristics if a result is not returned. Whereas, before it would just return the result from the first heuristic that matched, even if it was nothing.